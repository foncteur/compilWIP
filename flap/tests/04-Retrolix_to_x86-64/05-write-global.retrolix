external printf

globals (x)
  lx : x <- copy "The universe is big, it's vast and complicated, and ridiculous and sometimes, very rarely, impossible things just happen and we call them miracles.";
  l00: %rdi <- copy "%s\n";
  l01: %rsi <- copy x;
  l02: %rax <- copy 0;
  l03: printf();
  lx3: x <- copy "Like an assignment that works, for instance!";
  l04: %rdi <- copy "%s\n";
  l05: %rsi <- copy x;
  l06: %rax <- copy 0;
  l07: printf();
  l08: exit;
end
