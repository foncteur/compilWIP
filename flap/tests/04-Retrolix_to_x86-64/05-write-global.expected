Process exited with status exited(0).
STDOUT:
The universe is big, it's vast and complicated, and ridiculous and sometimes, very rarely, impossible things just happen and we call them miracles.
Like an assignment that works, for instance!

STDERR:



